src/reducer/filterReducer.js

```diff
@@ -21,6 +21,21 @@ const filterReducer = (state, action) => {
     case "WALKING":
       return { ...state, walking: action.payload };
 
+    case "4_STAR_&_ABOVE":
+      return { ...state, rating: "4_STAR_&_ABOVE" };
+
+    case "3_STAR_&_ABOVE":
+      return { ...state, rating: "3_STAR_&_ABOVE" };
+
+    case "2_STAR_&_ABOVE":
+      return { ...state, rating: "2_STAR_&_ABOVE" };
+
+    case "1_STAR_&_ABOVE":
+      return { ...state, rating: "1_STAR_&_ABOVE" };
```

Comment:
you can optimize this rating filter into a single function by using the `value` attributes of the radio button.